src/snowflake/snowpark/exceptions.py

```diff
@@ -79,12 +79,16 @@ def __init__(
         error_code: Optional[str] = None,
         sfqid: Optional[str] = None,
         query: Optional[str] = None,
+        sql_error_code: Optional[int] = None,
+        raw_message: Optional[str] = None,
```

Comment:
what will be the difference between `error_code` and `sql_error_code`, `message` and `raw_message`?
and who will be using(set/get) the two new error fields?

I ask because I feel those are duplicate to the existing ones
or maybe consider introducing a new sub class dedicate to your error scenario

Reply:
For example, if I get a [SQL_INTERNAL_ERROR](https://github.com/snowflakedb/snowflake/blob/f54a602c44a594c446b9803a0bbef042c3aec69c/GlobalServices/modules/platform/core/src/main/resources/com/snowflake/gs_error_messages.properties#L23), this is what each of the values will be:

* `error_code`: 1304 (I'm not entirely sure what this number means, but it looks like a Snowpark error code system distinct from the SQL error code system)
* `sql_error_code`: 000603
* `message`: Something like `000603 (P0000): SQL execution internal error: blah blah`
* `raw_message`: `SQL execution internal error: blah blah`

> and who will be using(set/get) the two new error fields?

The intent is only for this to be populated by the `snowpark-python` library when an error is thrown by `snowflake-connector` library. From what I can tell, `ProgrammingError` and `OperationalError` are the only two errors that can come from the connector library.